contracts/governance/Staking/Staking.sol

```diff
@@ -774,4 +957,48 @@ contract Staking is
                 block.number;
         require(notSameBlock, "S20"); //S20 : "cannot be mined in the same block as last stake"
     }
+
+    /**
+     * @notice Max iteration for direct withdrawal from staking to prevent out of gas issue.
+     *
+     * @return max iteration value.
+     */
+    function getMaxVestingWithdrawIterations() public view returns (uint256) {
+        return maxVestingWithdrawIterations;
+    }
+
+    /**
+     * @dev set max withdraw iterations.
+     *
+     * @param maxIterations new max iterations value.
+     */
+    function setMaxVestingWithdrawIterations(uint256 maxIterations) external onlyAuthorized {
+        require(maxIterations > 0, "S21");
+        uint256 oldIterations = maxVestingWithdrawIterations;
+        maxVestingWithdrawIterations = maxIterations;
+
+        emit MaxVestingWithdrawIterationsUpdated(oldIterations, maxVestingWithdrawIterations);
+    }
+
+    /**
+     * @dev get vesting creation type.
+     *
+     * @param duration duration of the vesting
+     *
+     * @return vesting creation type.
+     * If months not 10 | 26 | 48, it will return 0 as the type
+     */
+    function _getVestingCreationType(uint256 duration) internal pure returns (uint256) {
```

Comment:
pls add a mapping (duration => creationType), onlyAuthorized setter & tests to avoid extra SIP should we need to add another vesting contract type
also will need to add & run setter to initialize values - pls prepare relevant multisig tx creation script

Reply:
Moved it to storage, added the setter, and the script